.github/workflows/update.yml

```diff
@@ -0,0 +1,50 @@
+name: Update index.js
+
+on:
+  push:
+    paths:
+      - '.github/workflows/update.yml'
+  pull_request:
+    paths:
+      - '.github/workflows/update.yml'
+  schedule:
+    - cron: '11 22 * * *'
+  workflow_dispatch:
+
+jobs:
+  update:
+    if: ${{ github.event_name == 'schedule' || github.repository == 'ruby/setup-ruby' }}
+    name: update
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set ENV
+        run: |
+          echo "TODAY=$(date +%F)" >> $GITHUB_ENV
+      - name: Clear stored yarn data
+        run: >-
+          rm -fr
+          ~/.yarn
+          ~/.yarnrc
+          ~/.cache/yarn
+          ~/.config/yarn
+      - run: yarn upgrade
+      - run: yarn run package
+      - name: Check diffs
+        id: diff
+        run: git diff --no-ext-diff --ignore-submodules --stat --exit-code
+        continue-on-error: true
+      - name: Commit
+        run: |
+          git pull --ff-only origin ${GITHUB_REF#refs/heads/}
+          git add --all
+          git commit --message="Update packages at ${TODAY}"
+          git push origin ${GITHUB_REF#refs/heads/}
```

Comment:
I don't think I would want something that pushes automatically to master, but rather it creates a PR or something.

Also it means the permissions cannot just be:
```
permissions:
  contents: read
```
and so it's a lot harder to assess how safe is this or not (it seems pretty unsafe at first glance, any dep update could read the GitHub token etc).